### ISSUE TITLE
Allow rng to be int in exposed method

### ISSUE THREAD

**Issue by fehiepsi** [pyro-ppl/numpyro#372]
I think the pattern at [this PR](https://github.com/pyro-ppl/numpyro/pull/350/files#diff-c5378ba2ea50a6f7e17e51b65c5f251dR346) can be helpful for other utilities/classes such as `SVI.init`, `MCMC.run`, `AutoGuide.sample_posterior`, `predictive`, `Distribution.sample`. What do you think @neerajprad ?

**Comment by neerajprad** [comment]
I think it is fine to do it only if the `rng` is fed into the `seed` handler. Right now, our distribution's first argument matches that of JAX samplers. Since the latter requires the first argument to be a `PRNGKey`, it is nice if we can match this (soft) interface. However, many a time when we pass in `rng` to inference classes, we do things like `split` it or pass it to some distribution, etc. In all these cases we will have to wrap over `split` and `Distribution.sample` to do this safely.

The reason why this is useful for `seed` is because we can then use this in `pyro.generic` without having to import `jax.random` in Pyro. Otherwise, I do not see too much benefit in doing this apart from some code conciseness. It will just add another indirection layer and specially for `Distribution.sample` methods which might be called many more times (without jit) as compared to seed's init, this may add some small overhead. Let us keep this issue open, and add any use cases where this will be a clear win.

**Comment by fehiepsi** [comment]
Thanks for sharing your thought! It totally makes sense to me. Actually I want to close this issue because my proposal does not make sense to me anymore. Let’s revisit this if needed in the future.
